src/runtime/print_nintendoswitch.go

```diff
@@ -0,0 +1,14 @@
+// +build nintendoswitch
+
+package runtime
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+//go:nobounds
+func printstring(s string) {
+	sh := (*reflect.StringHeader)(unsafe.Pointer(&s))
```

Comment:
It can't be handled by putchar since the SVC requires a full string. Otherwise, this happens:
![image](https://user-images.githubusercontent.com/578310/81822611-b9a40580-9509-11ea-8945-e8402c90755b.png)

Reply:
Ah I see. But even if you special-case `printstring`, other print functions will still print variables vertically.
What we did in WebAssembly was to buffer a string until a line terminator and then print the entire string using `console.log`. That is done in JavaScript, but you can do something similar in Go. Something like:

```go
var stdoutBuffer = make([]byte, 0, 80)

func putchar(c byte) {
    if c == '\n' {
        // print entire string...
        stdoutBuffer = stdoutBuffer[:0]
        return
    }
    stdoutBuffer = append(stdoutBuffer, c)
}

Reply:
the write call on libnx is actually the write from newlib. It just writes the data to the specified file descriptor. But the stdout/stderr/stdin needs to be pre-initialized to point somewhere, otherwise they're closed.

Since this print calls are internal to the engine (for example for panics), I use the System Call  OutputDebugString to output to Emulator Terminal.

I will try the solution of buffering the output.

Reply:
![image](https://user-images.githubusercontent.com/578310/81880469-f866a980-9563-11ea-8c9e-6f2d00a21278.png)
Buffering works fine.